### DISCUSSION TITLE
Ability that allows players to "auto-turn" to targets by pressing a button or by game settings.

### DISCUSSION THREAD

**Discussion by Daywalkrr** [AscensionGameDev/Intersect-Engine#1530] [Ideas]
**Is your feature request related to a problem? Please describe.**
Aiming projectiles is a nightmare. All too often you end up moving around when you don't mean to.

**Describe the solution you'd like**
Introduce functionality where:
- A player can enable a "turn-to-target" option that turns them to face whatever target they've selected whenever a new target is determined
- A button on the keyboard that turns the player toward their current target

**Comment by Cheshire92** [comment]
Duplicate of your own issue https://github.com/AscensionGameDev/Intersect-Engine/issues/987

**Comment by Daywalkrr** [comment]
> 
> 
> Duplicate of your own issue #987

TBF, that issue is just for turning controls. I separated the issues because introducing a button that changes a character's direction is pretty trivial, but this ticket (an aiming system) is more open and up for debate as to how it should work.

**Comment by Arufonsu** [comment]
> TBF, that issue is just for turning controls. 

Which is indeed meant for something entirely different (turn around with ease in tile), so i'm reopening this one, i hope that's okay. Also, we should probably wait and see how #1016 does before we jump into this,

I personally like the idea of pressing a key (tab perhaps?) once in order to automatically aim / switch targets nearby (like LoZ games), and press it twice (or press esc?) in order to stop auto look at target (targeting).

**Comment by Justn53** [comment]
Hi. If not auto-aim (not a fan personally) why not move the block feature off of the mouse right-click and have holding right-click + the cursor position will aim/turn the player. Seems like it would well for both projectiles and melee. I'm not sure if my suggestion belongs here or in #987 

 I have seen other 2d games do it like this ( I cant think of any right now other than 'Project Zomboid').

**Comment by Arufonsu** [comment]
Not an issue anymore, but a discussion it seems.
I've worked around with @AlexVild 's face to target implementation a bit, and i can open up a PR as proposal for this to be added within the upcoming targeting game settings category as toggle-able ; as well with the ability to enable/disable it thru the server side since this is a game functionality changer.

**Comment by Arufonsu** [comment]
Opened a feat pr for this 👍🏽 
#1588

**Comment by Arufonsu** [comment]
Closed by #1588
